dungeon.gmx/objects/obj_boss.object.gmx

```diff
@@ -134,7 +357,51 @@ alarm[0] = 15;
         <arguments>
           <argument>
             <kind>1</kind>
-            <string>health_player -=10;
+            <string>//health_player -=10;
+image_speed = .5;
+if global.SFX and !audio_is_playing(SpaceWizardHammer)
+{
+    global.SFX = audio_play_sound(SpaceWizardHammer, 10, false);
+}
+if facing == 0 {
+    health_player -=1;
+    sprite_index = spr_boss_atkh_right;
+} else if facing == 1 {
+    //up/down
+    health_player -=1;
+    sprite_index = spr_boss_atkh_forward;
+} else if facing == 2 {
+    health_player -=1;
+    sprite_index = spr_boss_atkh_left;
+}
+
+</string>
+          </argument>
+        </arguments>
+      </action>
+      <action>
+        <libid>1</libid>
+        <id>603</id>
+        <kind>7</kind>
+        <userelative>0</userelative>
+        <isquestion>0</isquestion>
+        <useapplyto>-1</useapplyto>
+        <exetype>2</exetype>
+        <functionname></functionname>
+        <codestring></codestring>
+        <whoName>self</whoName>
+        <relative>0</relative>
+        <isnot>0</isnot>
+        <arguments>
+          <argument>
+            <kind>1</kind>
+            <string>with (other)
+{
+    direction = point_direction(other.x, other.y, x, y); //note that since with() pushes the code over to other, it is now the enemy that is "other" within that construct
```

Comment:
This is the code that causes the player to get pushed away, right? Not sure how this is causing the player to skip right into the wall, as it's just like any other movement, right? Thus, you'd think the player would just collide with the wall as usual.